### DISCUSSION TITLE
While Testing the inputfield , "TypeError: O is not a function "  error is showing in react native

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5496] [Q&A]
Can you help me to resolve this?

` FAIL  src/components/helpers/inputfields/__test__/inputfield.test.tsx (23.804 s)
  ● input field test › textField render

    TypeError: O is not a function

      124 |             onFocus={e => setActive(true)}
      125 |             onBlur={onBlur}
    > 126 |             onChangeText={text => onChange(text)}
          |                                   ^
      127 |             keyboardType={keyboardType}
      128 |             passwordRules={passwordrules}
      129 |             secureTextEntry={secure}

      at onChange (node_modules/react-hook-form/src/useController.ts:93:7)
      at onChangeText (src/components/helpers/inputfields/InputField.tsx:126:35)
      at node_modules/@testing-library/react-native/build/fireEvent.js:94:19
      at batchedUpdates (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:13452:12)
      at act (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:15280:14)
      at invokeEvent (node_modules/@testing-library/react-native/build/fireEvent.js:93:20)
      at Function.changeText (node_modules/@testing-library/react-native/build/fireEvent.js:103:49)
      at Object.<anonymous> (src/components/helpers/inputfields/__test__/inputfield.test.tsx:48:15)`

Testing code 
```javascript 
  const CustomComponent = () => {
    const methods = useForm();
    return (
      <FormProvider {...methods}>
        <InputField
          onError={methods.formState.errors?.Name}
          testID={'input'}
          width={43}
          height={34}
          bordercolor={'#343'}
          placeholder={'hello adify'}
          name="Name"
          controls={methods.control}
          rules={{required: false}}
        />
      </FormProvider>
    );
  };
  it('textField render', () => {
    const {getByTestId} = render(<CustomComponent />);
    fireEvent.changeText(getByTestId('input'), 'banana');
  });
});
```

this is the component
```javascript
  <Controller
        rules={rules}
        control={controls ? controls : control}
        name={name ? name : 'default'}
        render={({field: {onChange, onBlur, value}, fieldState, formState}) => (
          <TextInput
            style={{
              borderColor: bordercolorSet,
              width: width ? width : 312,
              height: height ? height : 46,
              color: '#153059',
              borderWidth: 1,
              borderRadius: borderradius ? borderradius : 0,
              paddingLeft: 16,
              fontSize: 14,
              fontFamily: 'Inter-Regular',
            }}
            placeholder={placeholder ? placeholder : 'Enter Some thing'}
            onFocus={e => setActive(true)}
            onBlur={onBlur}
            onChangeText={text => onChange(text)}
            keyboardType={keyboardType}
            passwordRules={passwordrules}
            secureTextEntry={secure}
            autoCorrect={false}
            maxLength={maxlength}
            numberOfLines={numberoflines}
            onEndEditing={onendediting}
            placeholderTextColor={placeholdertextcolor}
            ref={ref}
            testID={testID}
            textAlign={textalign}
            textAlignVertical={textalignvertical}
            value={value}
                   />
        )}
      />
```

**Comment by bluebill1049** [comment]
can you reproduce this in a codesandbox?
